src/rust/kafka/src/lib.rs

```diff
@@ -528,5 +584,12 @@ impl RetryProcessor {
                     result.map(|_| ())
                 }
             })
+            .then(|result| async {
+                if let Ok(_) = result {
+                    Ok(self.consumer.commit()?)
+                } else {
+                    result
+                }
```

Comment:
i think `result.and_then(|| self.consumer.commit())` would do the trick but it's fine as is

Reply:
done